tests/unit/src/test/scala/tests/LogBackupSuite.scala

```diff
@@ -0,0 +1,93 @@
+package tests
+
+import java.nio.file.Files
+
+import scala.util.control.NonFatal
+
+import scala.meta.internal.io.PathIO
+import scala.meta.internal.metals.Directories
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.MetalsServerConfig
+import scala.meta.internal.metals.RecursivelyDelete
+import scala.meta.internal.metals.logging.MetalsLogger
+import scala.meta.internal.metals.utils.LimitedFilesManager
+import scala.meta.io.AbsolutePath
+
+class LogBackupSuite extends BaseSuite {
+  val maxLogBackups = 2
+  val serverConfig: MetalsServerConfig =
+    MetalsServerConfig.default.copy(maxLogFileSize = 100, maxLogBackups = 2)
+  var workspace: AbsolutePath = _
+  def log: AbsolutePath = workspace.resolve(Directories.log)
+  def backupDir: AbsolutePath =
+    workspace.resolve(".metals").resolve(".backup_logs")
+  def limitedFilesManager = new LimitedFilesManager(
+    backupDir.toNIO,
+    maxLogBackups,
+    "log_",
+  )
+
+  override def beforeEach(context: BeforeEach): Unit = {
+    workspace = createWorkspace(context.test.name)
+    cleanWorkspace()
+  }
+
+  test("too-small-for-backup") {
+    log.writeText(".")
+    MetalsLogger.backUpOldLogFileIfTooBig(workspace, serverConfig)
+    assert(!backupDir.exists)
+  }
+
+  test("backup") {
+    log.writeText(List.range(1, 100).mkString)
+    MetalsLogger.backUpOldLogFileIfTooBig(workspace, serverConfig)
+    assert(backupDir.exists)
+    assert(limitedFilesManager.getAllFiles().size == 1)
+    assert(!log.exists)
+  }
+
+  test("backup-and-delete-overflow") {
+    def createBackupLog(text: String) =
+      backupDir.resolve(s"log_${System.currentTimeMillis()}").writeText(text)
+    val logdata = List.range(1, 100).mkString
+    log.writeText(logdata)
+    createBackupLog("1")
+    Thread.sleep(2)
+    createBackupLog("2")
+    Thread.sleep(2)
+    MetalsLogger.backUpOldLogFileIfTooBig(workspace, serverConfig)
+    assertNoDiff(
+      limitedFilesManager
+        .getAllFiles()
+        .sortBy(_.timestamp)
+        .map(file => Files.readString(file.toPath))
```

Comment:
Maybe just a small remark, it seems that `Files.readString` breaks JDK8 unit tests (method not available in Java 8). :+1: